package.json

```diff
@@ -7,7 +7,7 @@
     "test": "test"
   },
   "scripts": {
-    "test": "npm run lint && mocha --recursive test",
+    "test": "npm run lint && mocha --exit",
```

Comment:
I don't think we should include this change. The error you're experiencing is expressed in the `reconfigure using existing configuration` ReplSet test, and indicates that some `MongoClient` is not actually being torn down properly. I'll look  more into this now, but in the meantime I would just remove this change from this PR.

Another note, while you're spending time looking at these tests and using `co`: as you've noted `co` returns promises now, and `mocha` automatically handles promises (and their rejections) when they are the return value of a test. We have a bunch of tests that look like:
```
it('is a test', function(done) {
  co(function*() {
    // do stuff
    done();
  }).catch(function(err) {
    console.log(err.stack);
  });
});
```

which should be converted to:
```
it('is a test', function() {
  return co(function*() {
    // do stuff
  })
});
```

(NOTE: returning `co` Promise -and- no longer using `done`)
You could even use something like `co-mocha`, but I think we shouldn't commit too strongly to `co`, anticipating `async/await` in the future.